api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java

```diff
@@ -83,63 +84,76 @@
      * @return the map containing the properties in this config source
      */
     default Map<String, String> getProperties() {
-        Map<String, String> props = new HashMap<>();
+        final Map<String, String> props = new HashMap<>();
         getPropertyNames().stream().forEach((prop) -> props.put(prop, getValue(prop)));
         return props;
     }
 
     /**
-     * Gets all property names known to this config source, potentially without evaluating the values.
+     * Gets all property names known to this config source, potentially without
+     * evaluating the values.
      *
-     * For backwards compatibility, there is a default implementation that just returns the keys of {@code getProperties()}
-     * slower ConfigSource implementations should replace this with a more performant implementation
+     * For backwards compatibility, there is a default implementation that just returns
+     * an empty set. This method should be overriden.
```

Comment:
Minor, but I would prefer this documentation to read something like the following:

    The default implementation of this method returns an empty {@link Set}.
    {@link ConfigSource} implementations should override this method.

Reply:
done